sdk/openai/openai/test/public/utils/utils.ts

```diff
@@ -0,0 +1,203 @@
+      }
+      logger.warning(`Error in deployment ${deployment}: ${errorStr}`);
+      errors.push(errorStr);
+    }
+  }
+  if (errors.length > 0) {
+    throw new Error(`Errors list: ${errors.join("\n")}`);
+  }
+  assert.isNotEmpty(succeeded, "No deployments succeeded");
+  logger.info(`Succeeded with (${succeeded.length}): ${succeeded.join(", ")}`);
+  return succeeded;
+}
+
+export async function sendRequestWithRecorder(request: PipelineRequest): Promise<PipelineResponse> {
+  const client = createDefaultHttpClient();
+  const pipeline = createEmptyPipeline();
+  return pipeline.sendRequest(client, request);
+}
+
+async function listDeployments(
+  subId: string,
+  rgName: string,
+  accountName: string,
+): Promise<string[]> {
+  const deployments: string[] = [];
+  const mgmtClient = new CognitiveServicesManagementClient(createTestCredential(), subId);
+  for await (const deployment of mgmtClient.deployments.list(rgName, accountName)) {
+    const deploymentName = deployment.name;
+    if (deploymentName) {
+      deployments.push(deploymentName);
+    }
+  }
+  // console.log(`Available deployments (${deployments.length}): ${deployments.join(", ")}`);
+  return deployments;
+}
+
+export function updateWithSucceeded(succeeded: string[], deployments: string[]): void {
+  if (deployments.length === 0) {
+    deployments.push(...succeeded);
+  }
+}
+
+export function getSucceeded(deployments: string[], succeededDeployments: string[]): string[] {
+  if (succeededDeployments.length > 0) {
+    return succeededDeployments;
+  }
+  return deployments;
+}
+
+function getAccountNameFromResourceType(deploymentType: DeploymentType): string {
+  switch (deploymentType) {
+    case "completions":
+      return assertEnvironmentVariable(
+        EnvironmentVariableNamesForCompletions.ACCOUNT_NAME_COMPLETIONS,
+      );
+    case "dalle":
+      return assertEnvironmentVariable(EnvironmentVariableNamesForDalle.ACCOUNT_NAME_DALLE);
+    case "whisper":
+      return assertEnvironmentVariable(EnvironmentVariableNamesForWhisper.ACCOUNT_NAME_WHISPER);
+  }
+}
+
+export async function getDeployments(deploymentType: DeploymentType): Promise<string[]> {
+  return listDeployments(
+    assertEnvironmentVariable(EnvironmentVariableNamesAzureCommon.SUBSCRIPTION_ID),
+    assertEnvironmentVariable(EnvironmentVariableNamesAzureCommon.RESOURCE_GROUP),
+    getAccountNameFromResourceType(deploymentType),
+  );
+}
+
+export async function bufferAsyncIterable<T>(iter: AsyncIterable<T>): Promise<T[]> {
+  const result: T[] = [];
+  for await (const item of iter) {
+    result.push(item);
+  }
+  return result;
+}
+
+export async function get(url: string): Promise<PipelineResponse> {
+  const request = createPipelineRequest({
+    url,
+    method: "GET",
+    headers: createHttpHeaders(),
+    streamResponseStatusCodes: new Set([200]),
+  });
+  return sendRequestWithRecorder(request);
+}
+
+// TODO: Update this once we add Azure specific feature tests
```

Comment:
Is this still a relevant comment?

Reply:
Yes, I updated the output type to Azure extension